src/CommunityToolkit.Maui.BindablePropertySG/BindablePropertySG.cs

```diff
@@ -0,0 +1,192 @@
+	public string CoerceValueMethodName { get; set; } = string.Empty;
+	public string DefaultValueCreatorMethodName { get; set; } = string.Empty;
+
+	public BindablePropertyAttribute(string propertyName)
+	{
+		PropertyName = propertyName;
+	}
+}
+""";
+
+	public void Initialize(IncrementalGeneratorInitializationContext context)
+	{
+		context.RegisterPostInitializationOutput(ctx => ctx.AddSource("BindablePropertyAttribute.g.cs", SourceText.From(bpAttribute, Encoding.UTF8)));
+
+		var provider = context.SyntaxProvider.ForAttributeWithMetadataName("CommunityToolkit.Maui.BindablePropertySG.BindablePropertyAttribute`1",
+			SyntaxPredicate, SemanticTransform)
+			.Where(static x => x.ClassInformation != default || !x.BPInfos.IsEmpty)
+			.Collect()
+			.SelectMany(static (types, _) => types);
+
+
+		context.RegisterSourceOutput(provider, Execute);
+	}
+
+	void Execute(SourceProductionContext context, SemanticValues semanticValues)
+	{
+		var source = GenerateSource(semanticValues);
+		SourceStringService.FormatText(ref source);
+		context.AddSource($"{semanticValues.ClassInformation.ClassName}.g.cs", SourceText.From(source, Encoding.UTF8));
+	}
+
+	static string GenerateSource(SemanticValues value)
+	{
+		var sb = new StringBuilder($@"
+// <auto-generated>
+// Test2 : {DateTime.Now}
+
+namespace {value.ClassInformation.ContainingNamespace};
+
+{value.ClassInformation.DeclaredAccessibility} partial class {value.ClassInformation.ClassName}
+{{
+");
+
+		foreach (var info in value.BPInfos)
+		{
+			GenerateBindableProperty(sb, info);
+			GenerateProperty(sb, info);
+		}
+
+		sb.AppendLine().Append("}");
+		return sb.ToString();
+
+		static void GenerateBindableProperty(StringBuilder sb, BPInfo info)
+		{
+			/// <summary>
+			/// Backing BindableProperty for the <see cref="PropertyName"/> property.
+			/// </summary>
+			sb.AppendLine("/// <summary>")
+				.AppendLine($"/// Backing BindableProperty for the <see cref=\"{info.PropertyName}\"/> property.")
+				.AppendLine("/// </summary>");
+			
+			// public static readonly BindableProperty TextProperty = BindableProperty.Create(...);
+			sb.AppendLine($"public static readonly {bpFullName} {info.PropertyName}Property = ")
+							.Append($"{bpFullName}.Create(")
+							.Append($"\"{info.PropertyName}\", ")
+							.Append($"typeof({info.ReturnType}), ")
+							.Append($"typeof({info.DeclaringType}), ")
+							.Append($"{info.DefaultValue}, ")
+							.Append($"{bindingModeFullName}{info.DefaultBindingMode}, ")
+							.Append($"{info.ValidateValueMethodName}, ")
+							.Append($"{info.PropertyChangedMethodName}, ")
+							.Append($"{info.PropertyChangingMethodName}, ")
+							.Append($"{info.CoerceValueMethodName}, ")
+							.Append($"{info.DefaultValueCreatorMethodName}")
+							.Append(");");
+
+			sb.AppendLine().AppendLine();
+		}
+
+		static void GenerateProperty(StringBuilder sb, BPInfo info)
+		{
+
+			/// <inheritdoc />
+			sb.AppendLine("/// <inheritdoc />");
```

Comment:
For us this will work very well, not sure how to improve this when it goes public. This is inherit by the inline docs on the interface